### PR TITLE
Handle file mode for kubeconfig files

### PR DESCRIPTION
**Pull request by ananace**
Probably not the most optimal implementation, though it's best to at least ensure it's 0600 by default.

### CODE REVIEW

**Review by rwaffen — APPROVED** [review]

**Comment by ananace** [comment]
I don't suppose there's some vox pupuli module that's done file mode handling, since the code here is _very_ fragile.
I tried to look at the Puppet file handling itself, but it looked to be difficult to use outside of the file resource itself.

**Review by tuxmea — APPROVED** [review]
Looks OK to me. @ananace can you please verify that mode => 'u-x+rw,g-rwx,o-rwx' does also work?

**Comment by ananace** [comment]
@tuxmea I can verify that such a mode most definitely does *not* work, since the code has no concept except for octal modes at the moment. Hence my asking if there was any examples for using the mode parsing code from the Puppet file type outside of it, since I very much want to support mode strings as well.

**Comment by tuxmea** [comment]
@ananace on any puppet installation, check /opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/util/symbolic_file_mode.rb
This gets required by /opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/type/file/mode.rb and /opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/type/file.rb

**Comment by ananace** [comment]
@tuxmea Yes, those are the exact files I was asking about if anyone has actually used outside of said type. The symbolic file mode handling looks exactly like what I want after all, but they proved annoying to use in-place, and I wanted to avoid having to copy out Puppet code into module-local classes when they already exist.

File: lib/puppet/provider/kubeconfig/ruby.rb [hunk 1]
```diff
@@ -40,6 +42,13 @@ def chown
     FileUtils.chown(resource[:owner], resource[:group], resource[:path])
   end
 
+  def chmod
+    mode = resource[:mode]
+    mode = mode.to_i(8) if mode.is_a?(String) && mode =~ %r{^[0-7]+$}
```
Comment (bastelfreak) [inline lib/puppet/provider/kubeconfig/ruby.rb:47]: should we ensure we have 4 digits?

Comment (ananace) [reply lib/puppet/provider/kubeconfig/ruby.rb:47]: The munging and validation should only result in valid numerical strings, but I guess it doesn't hurt to make the validation stricter here as well.

**Comment by bastelfreak** [comment]
@ananace is this ready for merge?

**Comment by ananace** [comment]
It should be, I was debating the handling of multiple kubeconfig contexts - or contexts in files created elsewhere, but I figure it's very unlikely that the default of 0600 is going to cause any issues for people - since it's what the K8s tooling itself expects anyway.

**Review by rwaffen — APPROVED** [review]

**Pull request merged**